acceptanceTests/tap_test.go

```diff
@@ -10,10 +10,74 @@ import (
 	"os/exec"
 	"path"
 	"strings"
+	"sync"
 	"testing"
 	"time"
+
+	"github.com/gorilla/websocket"
+	"github.com/stretchr/testify/assert"
 )
 
+// waitTimeout waits for the waitgroup for the specified max timeout.
+// Returns true if waiting timed out.
+func waitTimeout(wg *sync.WaitGroup, timeout time.Duration) bool {
+	c := make(chan struct{})
+	go func() {
+		defer close(c)
+		wg.Wait()
+	}()
+	select {
+	case <-c:
+		return false // completed normally
+	case <-time.After(timeout):
+		return true // timed out
+	}
+}
+
+// checkDBHasEntries checks whether there are any entries in the database
+// before the given timestamp. Returns a slice of non-empty entries if it succeeds.
+func checkDBHasEntries(t *testing.T, timestamp int64, limit int) (entries []map[string]interface{}) {
+	query := fmt.Sprintf("timestamp < %d and limit(%d)", timestamp, limit)
+	webSocketUrl := getWebSocketUrl(defaultApiServerPort)
+
+	c, _, err := websocket.DefaultDialer.Dial(webSocketUrl, nil)
+	assert.Nil(t, err)
+	defer c.Close()
+
+	handleWSConnection := func(wg *sync.WaitGroup) {
+		defer wg.Done()
+		for {
+			_, message, err := c.ReadMessage()
+			if err != nil {
+				return
+			}
+
+			var data map[string]interface{}
+			err = json.Unmarshal([]byte(message), &data)
+			assert.Nil(t, err)
+
+			if data["messageType"] == "entry" {
+				entries = append(entries, data)
+			}
+		}
+	}
+
+	err = c.WriteMessage(websocket.TextMessage, []byte(query))
+	assert.Nil(t, err)
```

Comment:
same assert convention here
```suggestion
	if err != nil {
		t.Errorf("%v", err)
		return
	}
```